src/pages/index.js

```diff
@@ -20,100 +24,178 @@ import {
-const addModalTitle = addModalForm.querySelector("#add-modal-title");
-const addModalURL = addModalForm.querySelector("#add-modal-url");
-const addModalCloseButton = addModal.querySelector(".modal__close");
-
-/*-- Card Selector --*/
-const cardTemplate =
-  document.querySelector("#card-template").content.firstElementChild;
-const cardElement = cardTemplate.cloneNode(true);
-const cardListEl = document.querySelector(".card__list");
-const cardImageEl = cardElement.querySelectorAll(".card__image");
+const addModalSubmitButton = addModalForm.querySelector(".modal__button");
 
 /*-- Edit Modal --*/
 const editModal = document.querySelector("#edit-modal");
 const editModalForm = editModal.querySelector("#edit-modal-form");
-const editModalName = editModal.querySelector("#edit-modal-name");
-const editModalBio = editModal.querySelector("#edit-modal-bio");
-const editModalCloseButton = editModal.querySelector(".modal__close");
-
-/*-- Preview Modal --*/
-const previewModal = document.querySelector("#preview-modal");
-const previewModalCloseButton = document.querySelector("#preview-modal-button");
-const previewTitleEl = previewModal.querySelector("#modal-preview-title");
-const previewImageEl = previewModal.querySelector(".modal__preview-image");
+const editModalSubmitButton = editModalForm.querySelector(".modal__button");
 
 /*-- Profile Selectors --*/
+const profileAvatarForm = document.querySelector("#profile-modal-form");
+const profilePenIcon = document.querySelector(`#profile__avatar-pen-icon`);
 const profileEditButton = document.querySelector(".profile__button-edit");
 const profileAddButton = document.querySelector(".profile__button-add");
-const profileName = document.querySelector(".profile__name");
-const profileBio = document.querySelector(".profile__bio");
+const avatarModalSubmitButton =
+  profileAvatarForm.querySelector(".modal__button");
 
-const items = cardsData;
-const section = new Section({ items, renderer: createCard }, ".card__list");
-section.renderItems();
+/*-- Trash Icon Selectors --*/
+const trashModal = document.querySelector("#trash-modal");
+const trashModalSubmitButton = trashModal.querySelector(".modal__button");
 
 /*---------------------------------------------------*/
-/*                     Functions                     */
+/*                 Event Handlers                    */
 /*---------------------------------------------------*/
 
-function createCard(cardData) {
-  const name = cardData.name;
-  const altName = cardData.name;
-  const link = cardData.link;
-  const card = new Card(
-    { name, altName, link },
-    "#card-template",
-    handleImageClick
-  );
-  return card.getView();
+function handleAvatarFormSubmit(formValues) {
+  const profilePicObject = api.patchProfileAvatar(formValues.url);
+  changeSubmitButton(avatarModalSubmitButton);
+  profilePicObject.then((object) => {
+    avatarPopup.handleAvatarChange(object.avatar);
+  });
+  avatarPopup.close();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well